exe/slow_query_exporter

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env ruby
+
+require 'daemons'
+require 'gelf'
+require 'optparse'
+require 'syslog'
+require 'slow_query_exporter'
+
+RUN_DIRECTORY = "/var/run/slow_query_exporter"
+PROGRESS_FILE = "#{RUN_DIRECTORY}/last_timestamp"
+
+$graylog_host = "localhost"
+$graylog_port = 12201
+$delay = 0.1
+$verbose = false
+$foreground = false
+
+HELP_TEXT = "Usage: slow_query_exporter [-fv] [-d delay] [-h host] [-p port] slow_query.log
+Options:
+    -h, --host         The Graylog host
+    -p, --port         The Graylog port
+    -d, --delay        An interval to pause after each GELF message, in (possibly fractional) seconds
+    -v, --verbose      Print entries to stdout as they're parsed
+    -f, --foreground   Don't daemonize on startup
+    -?, --help         Display this help text
+"
+
+OptionParser.new do |opts|
+  opts.banner = HELP_TEXT
+
+  opts.on("-h", "--host") { |host| $graylog_host = host }
+  opts.on("-p", "--port") { |port| $graylog_port = port }
+  opts.on("-d", "--delay") { |delay| $delay = delay.to_f }
+  opts.on("-v", "--verbose") { $verbose = true }
+  opts.on("-f", "--foreground") { $foreground = true }
+  opts.on("-?", "--help") do
+    $stderr.puts(HELP_TEXT)
+    exit
+  end
+end.parse!
+
+if ARGV.empty?
+  $stderr.puts(HELP_TEXT)
+  exit 1
+end
+
+Daemons.daemonize(dir_mode: :normal, dir: RUN_DIRECTORY) unless $foreground
+Syslog.open("slow_query_exporter", Syslog::LOG_PID | Syslog::LOG_PERROR, Syslog::LOG_DAEMON)
+last_timestamp = begin
+                   IO.read(PROGRESS_FILE).to_i
+                 rescue Errno::ENOENT
+                   0
+                 end
+
+# We touch the logfile to make sure it exists before we start. Otherwise, "tail -F" will die.
+logfile = ARGV[0]
+File.open(logfile, "a") {}
+
+gelf = GELF::Notifier.new($graylog_host, $graylog_port, "WAN")
+parser = SlowQueryExporter::QueryParser.new
+
+begin
+  tail = IO.popen(["tail", "-F", "-n", "+0", logfile])
```

Comment:
That is fair. As far as I understand, `tail` will work on the file descriptor, not the name. I believe it's an irrelevant question given the name will not change as far as I can tell:

Anyway, from the man page:
> With --follow (-f), tail defaults to following the file descriptor, which means that even if a tail'ed file is renamed, tail will continue to track its end.  This default behavior is not desirable when you really want to track the actual name of the file, not the file descriptor (e.g., log rotation).  Use --follow=name in that case.  That causes tail to track the named file in a way that accommodates renaming, removal and creation.